### PR TITLE
docs(lua transform): Add a CSV parsing guide

### LINKED ISSUE #2247: Add a guide about parsing an unsupported data format using Lua transform
Vector's documentation could benefit from a guide describing how to parse structured logs in some textual data format, which is both not supported natively by Vector and is hard to be parsed reliably using regular expressions, using `lua` transform with a loadable third-party module.

### PR DESCRIPTION
**Pull request by a-rodin**
Closes #2247.

### CODE REVIEW

File: website/guides/advanced/parsing-csv-logs-with-lua.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+"""
+```
+
+With this `source` the `csv` module is loaded when Vector is started up (or if the `lua` transform is added later and the
+config is [automatically reloaded][docs.administration.process-management#automatic-reload-on-changes]) and can
+be used through the global variable `csv`.
+
+## Implementing Custom Parsing
+
+With `csv` module, the [`hooks.process`][docs.reference.transforms.lua#process] can be changed to the following:
```
Comment (binarylogic) [inline website/guides/advanced/parsing-csv-logs-with-lua.md:102]: With -> With the

**Comment by binarylogic** [comment]
@a-rodin what do you think about guide "assumptions" or "requirements" that would display at the top of every guide? These could be represented in the front-matter to provide consistent styling across all guides. Ex:

```
---
assumptions:
- You understand the basic Lua concepts.
- You understand the basic Vector concepts and understand how to set up a pipeline.
---

**Comment by a-rodin** [comment]
@binarylogic I've added assumptions to front matter. I've also extended the `lua` transform documentation with a section about data types, as it turned out to be required for the guide.

**Comment by binarylogic** [comment]
This looks great! I'll work on displaying the assumptions at the top of the guides. Otherwise :shipit:.

**Comment by binarylogic** [comment]
Excellent tutorial @a-rodin. I'm going to merge this so we can make it available.

**Pull request merged**